Feature/22

ChrisCottier opened the pull request:
Finished creating navigation element of driver registration. Button design do not match design docs, but the functionality is there.

l2silver commented on web-client/src/shared/components/drivers-registration/CurrentSlide/index.test.tsx:
```diff
@@ -0,0 +1,16 @@
+
+    const firstNameLabel = getByText('First Name');
+
+    expect(firstNameLabel).toBeInTheDocument;
```
Awesome!

l2silver commented on web-client/src/shared/components/drivers-registration/CurrentSlide/index.tsx:
```diff
@@ -0,0 +1,28 @@
+import PersonalInfo from '../PersonalInfo/index'
+
+interface Props {
+    currentSlide: number
```
perfect

l2silver commented on web-client/src/shared/components/drivers-registration/NavigationButtons/index.tsx:
```diff
@@ -0,0 +1,50 @@
+import React from 'react';
+import Button from '@material-ui/core/Button';
```
Perfect

l2silver reviewed (commented):
Hey, it looks like there are some lint warnings. I just changed the lint script to throw an error, but in the mean time, can you fix the lint warnings. And I'll try and figure out why netlify was angry.

ChrisCottier commented:
> Hey, it looks like there are some lint warnings. I just changed the lint script to throw an error, but in the mean time, can you fix the lint warnings. And I'll try and figure out why netlify was angry.

Added return types to React function components. Lint warnings are gone now.

l2silver commented on web-client/src/shared/components/drivers-registration/NavigationButtons/index.tsx:
```diff
@@ -0,0 +1,50 @@
+
+    return (
+        <>
+            <ButtonGroup size="large">
```
Perfect use of button group!

l2silver commented on web-client/src/shared/components/drivers-registration/CurrentSlide/index.test.tsx:
```diff
@@ -0,0 +1,16 @@
+
+describe('Current Slide', () => {
+  test('should render Personal Info slide first.', () => {
+    const { getByText } = render(<CurrentSlide currentSlide={1}/>);
```
Yes!!! Thank you for testing current slide and not just PersonalInfo.

l2silver reviewed (changes requested):
Sorry Chris, 1 more change and we're good.

ChrisCottier commented:
> Sorry Chris, 1 more change and we're good.

Sorry I'm not as familiar with GitHub reviews, I couldn't see what / where you wanted changes. Or is it related to the newlines at the end of the files?

l2silver commented on web-client/src/shared/components/drivers-registration/NavigationButtons/index.tsx:
```diff
@@ -0,0 +1,50 @@
+                onClick={prevSlide}
+                >Back
+                </Button>
+                :<></> }
```
Do you think you could change instances 
condition ? Component : <></>
to
condition && Component
for when condition is boolean
and condition ? Component : null
when condition has the possibility of being the number 0

ChrisCottier replied:
Does that mean just change the React fragments to null instead? Or did you want me to change the turnery logic as well?

ChrisCottier replied:
Just tried it out- changing the fragments to null got rid of the material ui warning.

l2silver replied:
I think that's a good thing, no?

ChrisCottier replied:
Totally, was just making sure I followed. New commit just pushed to this branch.

l2silver commented:
Hey Chris, sorry my comment wasn't recorded. It's this one: https://github.com/diglit/volunteer-toronto-volunteer-drivers-app/pull/27#discussion_r559697576

OksanaSam reviewed (approved):
Looks really nice, great job Chris!
